### ISSUE TITLE
Cypress tests fail because of Gatsby's 404 page

### ISSUE THREAD

**Issue by fwielstra** [gatsbyjs/gatsby#12068]
## Summary

For some reason, when running our existing Cypress tests, Gatsby gives us the development 404 page with the message "`There's not a page yet at /__/`". I know this used to work up until last week or so. I just went back a couple weeks in Git history though, and now what I think used to work a couple of weeks ago is also no longer working. Going back to that point does open the cypress randomized port page for a second though, before going to `localhost:8000/__/` triggering the 404 page.

TL;DR: Does anyone know how to configure Gatsby and / or Cypress to not trigger the development 404 page?

**Comment by freiksenet** [comment]
Thank you for the report! Could you include a reproduction? We do run cypress in our end-to-end tests, so it should be working at least in theory :)

**Comment by fwielstra** [comment]
Hey @freiksenet, thanks for your quick reply; I've taken our repo and stripped out as much as possible to avoid noise, and pushed it to https://github.com/fwielstra/gatsby-cypress-404

Steps to start and reproduce; note that npm may work just as well, just need to go into the individual packages and run the start commands.

```
yarn install
# in one terminal:
yarn start
# in another one:
yarn open
```

Then start the single `index.ts` test. Expected: it opens a Cypress wrapper around the page and checks if the text "hello world" is there. Actual: Gatsby 404 development page.

Now that I've stripped away a lot of the code, something tells me it's probably something small, or a dependency gone weird. I have tried removing the `if (page.path.includes("404")) return;` line in `gatsby-node.js` but that just causes a compile error unfortunately.

Let me know if you have trouble reproducing things.

**Comment by zachfeldman** [comment]
@fwielstra @freiksenet we have the same exact problem with our app, we get the Gatbsy 404 page when running Cypress tests:
![image](https://user-images.githubusercontent.com/1437644/54468406-a3ae7f80-4762-11e9-98cb-96d71864795e.png)

Is this just due to a configuration problem? We're pretty new to Gatsby! Thanks for any ideas.

**Comment by morenoh149** [comment]
@fwielstra your minimal example doesn't run for me. Error
```
$ yarn open
yarn run v1.15.2
$ yarn workspace cypress cypress open
error Command "cypress" not found.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed.
Exit code: 1
Command: /usr/local/Cellar/node/11.12.0/bin/node
Arguments: /usr/local/Cellar/yarn/1.15.2/libexec/lib/cli.js cypress open
Directory: /Users/harrymoreno/programming/gatsby-cypress-404/packages/cypress
Output:

info Visit https://yarnpkg.com/en/docs/cli/workspace for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
```
$ yarn run cypress open
yarn run v1.15.2
error Command "cypress" not found.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
please add a readme and doublecheck you can run the repo yourself to get more help.

**Comment by zachfeldman** [comment]
Upgrading from 3.1.1 to 3.2.0 fixed this for us!

**Comment by zachfeldman** [comment]
I'd close it if I could!

**Comment by lannonbr** [comment]
@zachfeldman I can close this issue if it has been resolved by upgrading to 3.2.0. Is this the case?

**Comment by zachfeldman** [comment]
Yup! It was resolved by an upgrade.

**Comment by lannonbr** [comment]
Awesome. Closing as this was resolved
